fix broken links: change from .txt to .adoc

Botato300 opened the pull request:
## Changes
- Just change from .txt to .adoc

## Context
Some links like “/docs/howto/update-hook-example” redirected to URLs ending in .txt, which no longer exist. They now redirect to the same URLs ending in .adoc.
Issue associated: #2006

Botato300 committed:
fix broken links: change from .txt to .adoc

Some links like “/docs/howto/update-hook-example” redirected to URLs ending in .txt, which no longer exist. They now redirect to the same URLs ending in .adoc.

dscho reviewed (approved):
Looks correct!

I will need to re-run the GitHub workflow to re-generate the manual pages with this change. However, the result would conflict with #2007 at the moment, therefore I will hold off until that PR is merged, then go ahead and integrate this here PR so that the results will _not_ conflict.

dscho commented:
> I will need to re-run the GitHub workflow to re-generate the manual pages with this change.

For the record, I [did that a couple of days ago](https://github.com/dscho/git-scm.com/actions/runs/15212588446), but despite my local confirmation that the patch works, the workflow did not result in any changes :-(

To1ne commented:
> change from .txt to .adoc 

This gift keeps on giving. I'll try to have a look tomorrow.

dscho committed:
docs/howto/*: re-generate

The script determines that it does not need to do anything if the
`docs/howto/*.html` in question exists already. But we have to fix the
`.txt` extension, which is now the `.adoc` extension.

So, after `git rm external/docs/content/docs/howto/*.html`, I simply
re-built them via:

	REBUILD_DOC=v2.50.0 RERUN=1 \
	bundle exec ruby script/update-docs.rb git-worktree en

Signed-off-by: Johannes Schindelin <johannes.schindelin@gmx.de>

dscho commented:
> > I will need to re-run the GitHub workflow to re-generate the manual pages with this change.
> 
> For the record, I [did that a couple of days ago](https://github.com/dscho/git-scm.com/actions/runs/15212588446), but despite my local confirmation that the patch works, the workflow did not result in any changes :-(

I figured it out. The problem is this line: https://github.com/git/git-scm.com/blob/e3e589e185335fe5d965c36a65321b322da2e1bd/script/update-docs.rb#L598

That's the line responsible for _not_ re-generating the `/docs/howto/*` files: It first verifies the absence of the static file. But that file already exists, so the script thinks there is nothing to do anymore.
